api/maketdata.go

```diff
@@ -114,3 +115,42 @@ func getTickersHandler(storage storage.Market) func(c *gin.Context) {
 		ginutils.RenderSuccess(c, blockatlas.TickerResponse{Currency: md.Currency, Docs: tickers})
 	}
 }
+
+// @Summary Get charts data for a specific coin
+// @Id get_charts_data
+// @Description Get the charts data from an market and coin/token
+// @Accept json
+// @Produce json
+// @Tags charts
+// @Param coin query int true "coin id"
+// @Param token query string false "token id"
+// @Param currency query string false "the currency to show charts" default(USD)
+// @Success 200 {object} blockatlas.ChartData
+// @Router /v1/market/charts [get]
+func getChartsHandler(router gin.IRouter) {
+	var charts = marketdata.InitCharts()
```

Comment:
is there a way to init once and not every time you call it?

Reply:
It is closure. It is only called once when getChartsHandler() is called